### PR TITLE
git: Fix install from source for linux

### PR DESCRIPTION
**Pull request by lewis6991**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [ ] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

-----

### CODE REVIEW

**Comment by lewis6991** [comment]
Can someone tell me if the failure is related to this change. To me it looks unrelated.

File: Formula/git.rb [hunk 1]
```diff
@@ -22,6 +22,10 @@ class Git < Formula
     depends_on "expat"
     depends_on "zlib"
   end
+  
+  if OS.linux?
+    depends_on "linux-headers"
+  end
```
Comment (issyl0) [inline Formula/git.rb]:
```suggestion
  depends_on "linux-headers" unless OS.mac?
```

**Comment by iMichka** [comment]
Could you also post the error you are trying to solve? On which OS are you trying to build git?

**Comment by lewis6991** [comment]
I'm getting this error on rhe7:

```
> brew install git
==> Downloading https://www.kernel.org/pub/software/scm/git/git-2.26.0.tar.xz
Updating Homebrew...
Already downloaded: /devtools/linuxbrew/../.cache/downloads/7ffeebae3fe34a52e488122b78c9b90326530eea782747bd112c17ba842583a5--git-2.26.0.tar.xz
==> make install prefix=/devtools/linuxbrew/Cellar/git/2.26.0 sysconfdir=/devtools/linuxbrew/etc CC=gcc-5 CFLAGS= LDFLAGS= NO_TCLTK=1 NO_APPLE_COMMON_CRYPTO=1 OPENSSLDIR=/de
Last 15 lines from /home/lewrus01/.cache/Homebrew/Logs/git/01.make:
                 from packfile.h:4,
                 from fuzz-pack-headers.c:1:
/devtools/linuxbrew/include/bits/errno.h:24:26: fatal error: linux/errno.h: No such file or directory
compilation terminated.
In file included from /devtools/linuxbrew/include/errno.h:35:0,
                 from git-compat-util.h:178,
                 from cache.h:4,
                 from object-store.h:4,
                 from fuzz-pack-idx.c:1:
/devtools/linuxbrew/include/bits/errno.h:24:26: fatal error: linux/errno.h: No such file or directory
compilation terminated.
make: *** [Makefile:2387: fuzz-commit-graph.o] Error 1
make: *** Waiting for unfinished jobs....
make: *** [Makefile:2387: fuzz-pack-headers.o] Error 1
make: *** [Makefile:2387: fuzz-pack-idx.o] Error 1
```

`brew install linux-headers` fixes this.

**Comment by iMichka** [comment]
Please add `revision 1 unless OS.mac?` between these 2 lines:
```
sha256 "9ece0dcb07a5e0d7366a92b613b201cca11ae368ab7687041364b3e756e495d6"
head "https://github.com/git/git.git", :shallow => false
```

I prefer that we have a new revision to make clear something has changed in the dependencies.

**Comment by glensc** [comment]
I guess with the [change requested](https://github.com/Homebrew/linuxbrew-core/pull/19934#issuecomment-605524392) the `sha256` removal should also be omitted.

**Comment by iMichka** [comment]
Not in this case. Our linux CI needs the linux bottle sha256 to be removed, it is a little bit different than on the mac CI.

**Comment by iMichka** [comment]
Thanks @lewis6991 !

**Pull request merged**